client/core/trade.go

```diff
@@ -1438,83 +1437,102 @@ func (c *Core) swapMatchGroup(t *trackedTrade, matches []*matchTracker, errs *er
-	// unspecified amount of time, which gives the client an
-	// msgjson.TransactionUndiscovered error to signal to the client to try
-	// broadcasting again or check their asset backend connectivity before
-	// hitting the broadcast timeout (and being penalized).
-	var needsResolution bool
-	timeout := t.broadcastTimeout() / 4
-	if err := t.dc.signAndRequest(init, msgjson.InitRoute, ack, timeout); err != nil {
-		var msgErr *msgjson.Error
-		needsResolution = errors.As(err, &msgErr) && msgErr.Code == msgjson.SettlementSequenceError
-		errs.add("error sending 'init' message for match %s: %v", match.id, err)
-	} else if err := t.dc.acct.checkSig(init.Serialize(), ack.Sig); err != nil {
-		errs.add("'init' ack signature error for match %s: %v", match.id, err)
-	}
-
-	// Update the match db data with the swap details.
-	proof.Script = contract
-	if match.Match.Side == order.Taker {
-		proof.TakerSwap = coinID
-		match.SetStatus(order.TakerSwapCast)
-	} else {
-		proof.MakerSwap = coinID
-		match.SetStatus(order.MakerSwapCast)
-	}
-	if len(ack.Sig) != 0 {
-		auth.InitSig = ack.Sig
-		auth.InitStamp = encode.UnixMilliU(time.Now())
-	}
-	if err := t.db.UpdateMatch(&match.MetaMatch); err != nil {
-		errs.add("error storing swap details in database for match %s, coin %s: %v",
-			match.id, coinIDString(t.wallets.fromAsset.ID, coinID), err)
+func (c *Core) sendInitAsync(t *trackedTrade, match *matchTracker, coinID, contract []byte) {
+	if match.sendingInitAsync {
+		return
 	}
 
-	// With updated swap data, attempt match status resolution.
-	if needsResolution {
-		go c.resolveMatchConflicts(t.dc, map[order.OrderID]*matchStatusConflict{
-			t.ID(): {
-				trade:   t,
-				matches: []*matchTracker{match},
-			},
-		})
-	}
+	match.sendingInitAsync = true
+	c.log.Debugf("Sending 'init' request for match %s, contract coin %v (%s)",
+		match.id, coinIDString(t.fromAssetID, coinID), t.wallets.fromAsset.Symbol)
 
-	return errs.ifAny()
+	// Send the init request asynchronously.
+	go func() {
+		var err error
+		defer func() {
+			t.mtx.Lock()
+			match.sendingInitAsync = false
+			if err != nil {
+				corder := t.coreOrderInternal()
+				details := fmt.Sprintf("Error notifying DEX of swap for match %s: %v", match.id, err)
+				t.notify(newOrderNote(SubjectSwapError, details, db.ErrorLevel, corder))
+			}
+			t.mtx.Unlock()
+		}()
+
+		ack := new(msgjson.Acknowledgement)
+		init := &msgjson.Init{
+			OrderID:  t.ID().Bytes(),
+			MatchID:  match.id[:],
+			CoinID:   coinID,
+			Contract: contract,
+		}
+		// The DEX may wait up to its configured broadcast timeout, but we will
+		// retry on timeout or other error.
+		timeout := t.broadcastTimeout() / 4
+		err = t.dc.signAndRequest(init, msgjson.InitRoute, ack, timeout)
+		if err != nil {
+			err = fmt.Errorf("error sending 'init' message: %v", err)
+			var msgErr *msgjson.Error
+			if errors.As(err, &msgErr) && msgErr.Code == msgjson.SettlementSequenceError {
```

Comment:
I don't see how this would ever result in match conflict resolution since `err = fmt.Errorf("error sending 'init' message: %v", err)` would make `var msgErr *msgjson.Error; errors.As(err, &msgErr)` always false.  How about using `%w` but also not bothering to wrap it until just before the `return`

i.e. 

```go
	var msgErr *msgjson.Error
	if errors.As(err, &msgErr) && msgErr.Code == msgjson.SettlementSequenceError {
		c.resolveMatchConflicts(t.dc, map[order.OrderID]*matchStatusConflict{
			t.ID(): {
				trade:   t,
				matches: []*matchTracker{match},
			},
		})
	}
	err = fmt.Errorf("error sending 'redeem' message: %w", err)
	return
```